### DISCUSSION TITLE
Arguments of get_qc_mpo

### DISCUSSION THREAD

**Discussion by jtcantin** [block-hczhai/block2-preview#76] [Q&A] [answered]
Hello,

Is there documentation on the details of the different arguments of `get_qc_mpo` in `pyblock2/driver/core.py`?

The examples I have found so far only show basic usage, with the exception of one that adds implicit reordering, and the function itself doesn't have comments to elaborate on them.

Thank you!

**Comment by hczhai** [comment, marked as answer]
Hi, thanks for your interest in using the block2 package.

I have updated the documentation page and you can now find the explanation of the arguments of ``get_qc_mpo`` in https://block2.readthedocs.io/en/latest/api/pyblock2.html#pyblock2.driver.core.DMRGDriver.get_qc_mpo.

**Reply by jtcantin** [reply]
Thank you!

For g2e, which notation is assumed (i.e. chemist vs physicist) and are there any assumed prefactors (such as the 1/2 that often appears with the physicist notation)?

**Reply by hczhai** [reply]
The (unpacked) ``g2e`` uses chemists' notation, without any assumed prefactors (which is the notation used in [pyscf](https://github.com/pyscf/pyscf)).

In SU2 symmetry (spin restricted) mode, the Hamiltonian is given by
$$H = \sum_{\sigma,ij} [\mathrm{h1e}]\_{ij}\ a^{\dagger}\_{i\sigma} a_{j\sigma} + \frac{1}{2} \sum_{\sigma\sigma',ijkl} [\mathrm{g2e}]_{ijkl}\ a^{\dagger}\_{i\sigma} a^\dagger\_{k\sigma'} a\_{l\sigma'} a\_{j\sigma}$$

In SZ symmetry (spin unrestricted) mode, the Hamiltonian is given by
$$H = \sum_{\sigma,ij} [\mathrm{h1e}]\_{\sigma,ij}\ a^{\dagger}\_{i\sigma} a_{j\sigma} + \frac{1}{2} \sum_{\sigma\sigma',ijkl} [\mathrm{g2e}]_{\sigma\sigma',ijkl}\ a^{\dagger}\_{i\sigma} a^\dagger\_{k\sigma'} a\_{l\sigma'} a\_{j\sigma}$$

with $[\mathrm{h1e}][0] = [\mathrm{h1e}]\_{\alpha}, [\mathrm{h1e}][1] = [\mathrm{h1e}]\_{\beta}$ and $[\mathrm{g2e}][0] = [\mathrm{g2e}]\_{\alpha\alpha}, [\mathrm{g2e}][1] = [\mathrm{g2e}]\_{\alpha\beta}, [\mathrm{g2e}][2] = [\mathrm{g2e}]\_{\beta\beta}$.

In SGF symmetry (general spin) mode, the Hamiltonian is given by
$$H = \sum_{ij} [\mathrm{h1e}]\_{ij}\ a^{\dagger}\_{i} a_{j} + \frac{1}{2} \sum_{ijkl} [\mathrm{g2e}]_{ijkl}\ a^{\dagger}\_{i} a^\dagger\_{k} a\_{l} a\_{j}.$$

**Reply by jtcantin** [reply]
Thank you!
